server/src/main/java/org/elasticsearch/indices/recovery/plan/ShardRecoveryPlanners.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.indices.recovery.plan;
+
+import org.elasticsearch.index.store.Store;
+import org.elasticsearch.index.store.StoreFileMetadata;
+
+import java.util.List;
+
+public class ShardRecoveryPlanners implements ShardRecoveryPlanner {
```

Comment:
Related to my previous comment, I'd rather just have the logic to loop over different planners in the nontrivial version of the planner service instead of having a composite that ends up always being used like this.